ios/FluentUI/Vnext/Button/Button.swift

```diff
@@ -3,21 +3,101 @@
 //  Licensed under the MIT License.
 //
 
-import Combine
-import UIKit
 import SwiftUI
+import UIKit
+
+/// View that represents the button
+public struct MSFButtonView: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var tokens: MSFButtonTokens
+    @ObservedObject var state: MSFButtonStateImpl
+    var action: () -> Void
```

Comment:
Should the action be part of the state? I think that's how we've done other controls so far

Reply:
The challenge here is that we want to provide a sender parameter of type MSFButton to UIKit callers, but SwiftUI does not care about that and should be just an "() -> void" closure.

If we move the action to the state, we give up the control of providing the MSFButton sender param in the UIKit environment as the client code will be setting the closure directly and would need to capture the MSFButton instance on the closure event.

I think I'm okay with making the action closure simple and just have it "() -> void" everywhere, but this deserves its own separate PR.

Reply:
After we spoke offline, I've made the smaller change that does not affect the public State protocol. We'll iterate on a discussion to eventually remove the sender from the public API of the UIKit wrapper.
If that happens, we can then make the action a property of the state protocol.